File: app/src/main/java/org/jay/launchstarterexample/tasks/GetDeviceIdTask.java
package org.jay.launchstarterexample.tasks;

import android.provider.Settings;

import org.jay.appstarter.Task;
import org.jay.launchstarterexample.MyApplication;

public class GetDeviceIdTask extends Task {

    private String mDeviceId;

    @Override
    public void run() {
        MyApplication app = (MyApplication) mContext;
        mDeviceId = Settings.System.getString(mContext.getContentResolver(), Settings.System.ANDROID_ID);
        app.setDeviceId(mDeviceId);
    }

}

File: app/src/main/java/org/jay/appstarterexample/tasks/GetDeviceIdTask.kt
package org.jay.appstarterexample.tasks

import android.provider.Settings
import org.jay.appstarter.Task
import org.jay.appstarterexample.MyApplication

class GetDeviceIdTask : Task() {

    private var mDeviceId: String? = null

    override fun run() {
        val app = mContext as MyApplication
        mDeviceId =
            Settings.System.getString(mContext.contentResolver, Settings.System.ANDROID_ID)
        app.deviceId = mDeviceId
    }

}